plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageClientSettings.java

```diff
@@ -44,11 +48,35 @@
 
     /** A json Service Account file loaded from secure settings. */
     static final Setting.AffixSetting<InputStream> CREDENTIALS_FILE_SETTING = Setting.affixKeySetting(PREFIX, "credentials_file",
-        key -> SecureSetting.secureFile(key, null));
+            key -> SecureSetting.secureFile(key, null));
 
-    /** An override for the Storage endpoint to connect to. */
+    /**
+     * An override for the Storage endpoint to connect to. Deprecated, use host
+     * setting.
+     */
     static final Setting.AffixSetting<String> ENDPOINT_SETTING = Setting.affixKeySetting(PREFIX, "endpoint",
```

Comment:
Why does it need to be deprecated?

Reply:
The only reason is for the setting name to follow the rename of the sibling setting name in the library.
You can indeed set the protocol and port using the `host` setting, but, there are some cases when new connections don't follow the port specification. My impression is that the client library in fact expects this to only specify the hostname, yet it is not enforced or tested yet, and I think we should relay this "suggestion". No strong bias, though.

Reply:
I did, `endpoint` is the sole hostname setting .